client-sdk/ts-web/core/src/client.ts

```diff
@@ -0,0 +1,1341 @@
+    'NodeController',
+    'RequestShutdown',
+);
+const methodDescriptorNodeControllerWaitSync = createMethodDescriptorUnary<void, void>(
+    'NodeController',
+    'WaitSync',
+);
+const methodDescriptorNodeControllerIsSynced = createMethodDescriptorUnary<void, boolean>(
+    'NodeController',
+    'IsSynced',
+);
+const methodDescriptorNodeControllerWaitReady = createMethodDescriptorUnary<void, void>(
+    'NodeController',
+    'WaitReady',
+);
+const methodDescriptorNodeControllerIsReady = createMethodDescriptorUnary<void, boolean>(
+    'NodeController',
+    'IsReady',
+);
+const methodDescriptorNodeControllerUpgradeBinary = createMethodDescriptorUnary<
+    types.UpgradeDescriptor,
+    void
+>('NodeController', 'UpgradeBinary');
+const methodDescriptorNodeControllerCancelUpgrade = createMethodDescriptorUnary<
+    types.UpgradeDescriptor,
+    void
+>('NodeController', 'CancelUpgrade');
+const methodDescriptorNodeControllerGetStatus = createMethodDescriptorUnary<
+    void,
+    types.ControlStatus
+>('NodeController', 'GetStatus');
+const methodDescriptorDebugControllerSetEpoch = createMethodDescriptorUnary<types.longnum, void>(
+    'DebugController',
+    'SetEpoch',
+);
+const methodDescriptorDebugControllerWaitNodesRegistered = createMethodDescriptorUnary<
+    number,
+    void
+>('DebugController', 'WaitNodesRegistered');
+
+export class GRPCWrapper {
+    client: grpcWeb.AbstractClientBase;
+    base: string;
+
+    constructor(base: string) {
+        this.client = new grpcWeb.GrpcWebClientBase({});
+        this.base = base;
+    }
+
+    protected callUnary<REQ, RESP>(
+        desc: grpcWeb.MethodDescriptor<REQ, RESP>,
+        request: REQ,
+    ): Promise<RESP> {
+        // @ts-expect-error missing declaration
+        const name = desc.name;
+        return this.client.thenableCall(this.base + name, request, null, desc).catch((e) => {
+            if (e.message === 'Incomplete response') {
+                // This is normal. grpc-web freaks out if the response is `== null`, which it
+                // always is for a void method.
+                // todo: unhack this when they release with our change
+                // https://github.com/grpc/grpc-web/pull/1025
+                return undefined;
+            } else {
+                throw e;
+            }
+        });
+    }
+
+    protected callServerStreaming<REQ, RESP>(
+        desc: grpcWeb.MethodDescriptor<REQ, RESP>,
+        request: REQ,
+    ): grpcWeb.ClientReadableStream<RESP> {
+        // @ts-expect-error missing declaration
+        const name = desc.name;
+        return this.client.serverStreaming(this.base + name, request, null, desc);
+    }
+}
+
+export class NodeInternal extends GRPCWrapper {
+    constructor(base: string) {
+        super(base);
+    }
+
+    // beacon
+
+    /**
+     * GetBaseEpoch returns the base epoch.
+     */
+    beaconGetBaseEpoch() {
```

Comment:
We could also split these into multiple classes so instead of `nic.beaconGetBaseEpoch()` you would do `nic.beacon.getBaseEpoch()`?

Reply:
yeah I'll file this

Reply:
#39